### DISCUSSION TITLE
Add Custom Class to individual Heading Levels

### DISCUSSION THREAD

**Discussion by colinhall17** [ueberdosis/tiptap#2124] [General Questions] [answered]
Hi

I'm new to tiptap and so far loving it.

Is it possible to add custom CSS classes to individual Heading Levels.  I can see it can be done on a Heading as a whole but that applies to all levels.

For example I would like to have the following output:-

```
<h1 class="text-2xl">Header</h1>
<h2 class="text-xl">Header</h2>
```

Secondly, is it possible to toggle a class through commands/the chain?

For example:-

editor.command.toggleClass('custom-class')

or

editor.chain().addClass('custom-class').focus().run()

Thanks

**Comment by hanspagel** [comment, marked as answer]
1) See #1514 for a workaround!

2) You need to add a "class" attribute and update that attribute.
 
https://tiptap.dev/guide/custom-extensions#attributes
https://tiptap.dev/api/commands/update-attributes

**Reply by colinhall17** [reply]
Thanks, will look into

**Reply by christian-lorenz** [reply]
2. should be included in the docs. ;)
